### ISSUE TITLE
Received whats app audio message in firefox 2.5 is broken

### ISSUE THREAD

**Issue by sap-nocops** [loqui/im#871]
I've got a flame and I just upgrade the system to Firefox OS 2.5 (nightly channel).
With Firefox OS 2.2 everything was fine.
Now when I receive a whats app audio message I can download it but when I open it the music player opens but I can't reproduce the audio.
I tried to copy the audio file on my laptop and open it with vlc but no luck, it doesn't work with vlc either.

If you need any help or more information just tell me. I am a programmer by the way

**Comment by TitanNano** [comment]
can you reproduce the issue? did you receive an other file?

**Comment by sap-nocops** [comment]
Yes, I received like 4 or 5 five audio messages and for all of them I had this problem

**Comment by TitanNano** [comment]
okay, can you connect your device with the WebIDE and fetch a log while you
download a audio file?

2015-11-04 20:35 GMT+01:00 sap-nocops notifications@github.com:

**Comment by sap-nocops** [comment]
I've never done that before but I'll figure it out. Now I don't have time. I'll do that as soon as I could,
couple of hours I believe, and I'll let you know

**Comment by sap-nocops** [comment]
I can't see the log directly cause I need this:
"The device must be running a development build of Firefox OS 1.2+".
So I built loqui IM and installed it on my device but I got stuck because of #872

**Comment by TitanNano** [comment]
yes. Right now we can't figure out whats the problem. As soon as we find a solution we'll fix it.

**Comment by sap-nocops** [comment]
Here is the log when the audio message is received and played.

http://pastebin.com/wXfmkxpb

**Comment by sap-nocops** [comment]
Are there any news about this?
Can you reproduce the problem?
If you need more info or logs just ask

**Comment by TitanNano** [comment]
@sap-nocops sorry, we need a log where you actually download the file, not just receive the message.

**Comment by sap-nocops** [comment]
I enabled the write log to file option in the developer menu.
I downloaded an audio message that I already received.
here is the log

SAVING loqui/files/20151116T235726ZAV3HP.mp3
[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object] 211,12
SUCCESS
SAVING loqui/files/20151116T235726ZAV3HP.mp3
[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object] 211,12
SUCCESS
Write node called with  {"_tag":"presence","children":[],"attributes":{"type":"unavailable"},"_data":null} ~ self._writeNode@app://{6c55f296-5591-4a7a-be35-c1ddf808ba1b}/scripts/mozillahispano/coseme.js:7692:7 [ConnectionManager]
<presence type="unavailable">
</presence>
 ~ BinaryWriter.prototype._write@app://{6c55f296-5591-4a7a-be35-c1ddf808ba1b}/scripts/mozillahispano/coseme.js:4821:1 [BinaryWriter]
Received socket data: 23 bytes! ~ BinaryReader.prototype.onSocketData@app://{6c55f296-5591-4a7a-be35-c1ddf808ba1b}/scripts/mozillahispano/coseme.js:5259:5 [BinaryReader]
<presence from="393884609902@s.whatsapp.net" type="unavailable">
</presence>
 ~ BinaryReader.prototype._readNextTree@app://{6c55f296-5591-4a7a-be35-c1ddf808ba1b}/scripts/mozillahispano/coseme.js:5354:7 [BinaryReader]

Don't think it is useful

**Comment by TitanNano** [comment]
I need to do some tests, because I can't tell the issue from the logs. But it'll take me at least until Saturday next week... so if someone else can investigate here I wouldn't mind.

**Comment by cmeerw** [comment]
@sap-nocops Could you try downloading the file (from the URL in the log file) on your PC as well and compare the files downloaded via loqui with the one downloaded on the PC.

**Comment by sap-nocops** [comment]
Downloading the file from the url on my pc works fine, I can play it with vlc.
I can't play the one in the loqui/files folder on my phone though, neither from my pc with vlc.
I noticed that the downloaded one from the url is aac format, the one I find in the loqui/files folder on my phone is mp3 format.

**Comment by sap-nocops** [comment]
I found out that if I play the audio file I find in loqui/files on my phone with mplayer it works just fine
